onnxruntime/core/optimizer/graph_output_deduplication.cc

```diff
@@ -0,0 +1,109 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+// graph_output_deduplication.cc
+
+#include "core/optimizer/graph_output_deduplication.h"
+#include <queue>
+
+namespace onnxruntime {
+
+/*
+  Deduplication logic for graph outputs:
+  if this node produces a graph output which is a duplicate output, introduce an identity
+  node in front of each such duplicated output keeping the graph output order consistent
+  before and after transformation.
+
+  Example:
+
+    An Add op shown below with duplicate graph outputs as X
+                  Add
+                /    \
+              X       X
+
+    Transforms to:
+                  Add
```

Comment:
As walkaround, a better transform could be
```
     Add
     /\
    X  Identity
           \
           X1
```

In this way, it wont' break post processing that expected an output named X.

Reply:
Yeah, that makes sense. I can update the graph transformation such that the first output is same as the original one.